sdks/python/apache_beam/testing/test_utils_test.py

```diff
@@ -95,6 +95,43 @@ def test_cleanup_topics(self):
     pub_client.delete_topic.assert_called_with(topic=topic.name)
 
 
+class LCGeneratorTest(unittest.TestCase):
+  Generator = utils.LCGenerator
+
+  def test_generator_seed_results(self):
+    generator = self.Generator()
+    generator.seed(0)
+    self.assertEqual(generator.next_int(), 4232237)
+
+    generator.seed(1)
+    self.assertEqual(generator.next_int(), -1151252339)
+    self.assertEqual(generator.next_uint(), 3745583449)
+    self.assertAlmostEqual(generator.random_sample(), 0.375548, delta=1e-6)
+    self.assertEqual(generator.rand_bytes(10), b'\xa6\x8fW\xcb\xb1\xa88]dP')
+
+  def test_generator_seed_jdk_results(self):
+    generator = self.Generator()
+    generator.seed_jdk(0)
+    self.assertEqual(generator.next_int(), -1155484576)
+
+    generator.seed_jdk(1)
+    # the first next_int after seed_jdk(1) is close to seed_jdk(0)
+    self.assertEqual(generator.next_int(), -1155869325)
```

Comment:
just curious (you probably verified this) - does the value still match the JDK value after some large number of iteration? no need to make a test for this.

Reply:
Just did some more test:

Python code:

```python

from apache_beam.testing.synthetic_pipeline import get_generator

if __name__ == '__main__':
  gen = get_generator(algorithm='lcg')
  gen.seed_jdk(1)
  for i in range(10):
    print(gen.next_int())
```

Java code:
```java

public static void main(String[] argv) {
  Random rand = new Random(1);
  for (int i = 0; i < 10; ++i) {
    System.out.println(rand.nextInt());
  }
}
```

Both has result
```
-1155869325
431529176
1761283695
1749940626
892128508
155629808
1429008869
-1465154083
-138487339
-1242363800
```